`Accordion` - web docs (HDS-2035) (#1450)
hashicorp/design-system: website/docs/components/accordion/partials/guidelines/guidelines.md

> ## Usage

Reviewer: One thing I think is missing is an indication if it's acceptable or not to mix&match in the same accordion different types of accordion item (with/without interactive elements in the toggle).
Something like this, is OK? 
<img width="805" alt="screenshot_2880" src="https://github.com/hashicorp/design-system/assets/686239/3310ef6d-0b9a-4c76-aaaa-ce7c4b7157fd">

(no strong opinions, but as a consumer I would have a doubt and it would be better to clarify our guidance on this)

Author: @cveigt

Reviewer 2: Yes, this is a good point. I clarified our recommendation in the Generic content section. @KristinLBradley, could you update this example so that both containers `containsInteractive`?

### Accordion vs Reveal

While similar in functionality and interaction, the Accordion and Reveal are meant to be used in different scenarios and use cases.

#### When to use the Accordion

- You have multiple sections of content that can be expanded or collapsed.

- Each section of content has its own toggle content, giving users the ability to access specific sections.

- Users need to navigate through different sections of content and selectively expand or collapse sections

- There is a sequential or hierarchical relationship between the different sections.

#### When to use the Reveal

- You have a single piece of content that can be expanded to collapsed to reveal additional details or information.

- The content within the toggle is plain text.

- The content within the reveal is relatively short and simple.

- There is no need for multiple independent sections or hierarchical relationship between different sections.

### Usage examples

!!! Do

Use the Accordion when you need to display content in different sections that have a sequential or hierarchical relationship between them.

> ![Accordion with basic content](/assets/components/accordion/usage-do.png)

Reviewer: The images have all different sizes, which results in having the examples showcased all with different apparent sizes, which is weird

here the chevron sizes are different
<img width="173" alt="screenshot_2878" src="https://github.com/hashicorp/design-system/assets/686239/7c9f1dbe-d2c8-49b7-a00d-77db69f831c0">

here the text size is differeng
<img width="921" alt="screenshot_2876" src="https://github.com/hashicorp/design-system/assets/686239/b6b197ba-449a-489c-8c66-e3c4b6759ff6">

a simple trick is to use always the same **width** for the images (leaving white space if necessary) so they maintain the same ratio one in relation to the other (even when the page is resized). the alternative is to always set the width of the images in the markdown, which is much more work to do

Author: @cveigt

Reviewer 2: Updated!

> ![Accordion with basic content](/assets/components/accordion/usage-do.png)

Reviewer 3: ```suggestion
![Accordion with basic content](/assets/components/accordion/usage-do.png=800x*)
```

!!!

!!! Dont

Don’t use the Accordion when you need to display a single piece of content that can expand and collapse. Use the [Reveal](/components/reveal) instead.

> ![Accordion with basic content](/assets/components/accordion/usage-do-not.png)

Reviewer 3: ```suggestion
![Accordion with basic content](/assets/components/accordion/usage-do-not.png =610x*)
```

!!!

!!! Do

Use the Accordion when the toggle or the content is relatively complex.

> ![Accordion with complex content](/assets/components/accordion/usage-do-complex.png)

Reviewer 3: ```suggestion
![Accordion with complex content](/assets/components/accordion/usage-do-complex.png =800x*)
```

!!!

## Toggle content

The toggle accepts many different types of content, from text-based content to nested HDS components or custom content.

![Example of toggle content](/assets/components/accordion/toggle-content-text.png =690x*)

### Generic content

![Example of toggle content](/assets/components/accordion/toggle-content-custom.png =690x*)

### Interactive content

> Use `isInteractive` when nesting interactive content on the toggle.

Author: @cveigt

> Use `isInteractive` when nesting interactive content on the toggle.

Reviewer 3: +1 to @didoo's comment above. We should help consumers understand when to use this variant over the default one. 

```suggestion
Use `containsInteractive` when nesting interactive content in the toggle.
```

> Use `isInteractive` when nesting interactive content on the toggle.

Reviewer 2: I've expanded on this section with more and clearer guidance.

![Example of toggle content](/assets/components/accordion/toggle-content-interactive.png =690x*)

## Content type

!!! Info

The content type property is only relevant within Figma and doesn’t exist as a property in the code.
!!!

The `content` supports any custom content, local components, or Helios components via an `instance swap` property (customInstance) in Figma. In code, `yield` is supported.

### Default

![Example of content in accordion item](/assets/components/accordion/content-text.png =690x*)

### Custom

![Example of content in accordion item](/assets/components/accordion/content-custom.png =690x*)

## isOpen

In code, all accordion items are collapsed by default. Pass `isOpen` to display a single item open on page load.

> ![Example of content in accordion item](/assets/components/accordion/accordion-item-open.png =800x*)

Reviewer 3: This isn't necessarily usage guidance but more technical guidance for setting an open section in the code, which is already covered in the Code tab. We should probably remove this so it doesn't need to be maintained in two sections.

!!!

!!! Dont

Don’t use the Accordion when you need to display a single piece of content that can expand and collapse. Use the [Reveal](/components/reveal) instead.

![Accordion with basic content](/assets/components/accordion/usage-do-not.png =770x*)
!!!

!!! Do

Use the Accordion when the toggle or the content is relatively complex.

![Accordion with complex content](/assets/components/accordion/usage-do-complex.png =770x*)
!!!

## Toggle content

The toggle accepts many different types of content, from text-based content to nested HDS components or custom content.

![Example of toggle content](/assets/components/accordion/toggle-content-text.png =690x*)

### Generic content

![Example of toggle content](/assets/components/accordion/toggle-content-custom.png =690x*)

### Interactive content

Use `isInteractive` when nesting interactive content in the toggle. This lets users using assistive technology, access and interact with the nested element.

- Use this variant only when you need to put interactive elements such as a button or a link within the toggle area. Avoid placing complex elements that may compromise the usability and accessibility of the component.

> - When using this property, it should be applied to all items. Avoid mixing with default containers.

Reviewer: Should this be placed in a "warning" banner?

Author: @cveigt
